airflow/www/static/js/dag/Main.tsx

```diff
@@ -108,15 +110,21 @@ const Main = () => {
   const resize = useCallback(
     (e: MouseEvent) => {
       const gridEl = gridRef.current;
-      if (
-        gridEl &&
-        e.x > minPanelWidth &&
-        e.x < window.innerWidth - minPanelWidth
-      ) {
-        const width = `${e.x}px`;
-        gridEl.style.width = width;
-        saveWidth(width);
+      if (gridEl) {
+        if (e.x > minPanelWidth && e.x < window.innerWidth - minPanelWidth) {
+          const width = `${e.x}px`;
+          gridEl.style.width = width;
+          saveWidth(width);
+        } else if (
+          e.x < minPanelWidth &&
```

Comment:
What if `e.x === minPanelWidth`?

Reply:
So the logic states that if we are beyond `minPanelWidth` then adjust the resizer based on mouse movement and when we are lesser than `minPanelWidth` then given the mouse is moving right, snap to the `minPanelWidth` position. 

TL;DR --> Here, we care about what the mouse is doing when not at `minPanelWidth`. But the movement is tracked whether is at `minPanelWidth` or not.

Reply:
Should I also add the other way around i.e when a user tries to reduce the size beyond `minPanelWidth` then it should automatically collapse?

Reply:
Not a bad idea! Its certainly weird to have set width states and a variable draggable column, but if we can integrate them well then that would be best.

Reply:
Let's integrate it and see the UI experience. We need to have some minima anyways.

Reply:
Implemented the other way around as well. But have kept a heuristic where the user needs to cross `minPanelWidth/2` in order to collapse, since there would be cases where the user would reduce the size without intent to collapse. The intent here would be to cross the halfway mark. Let me know your opinion on this.

Reply:
Looks good! Mind to leave some comments for all this if/else logic? Then I think its good to merge

Reply:
Added!